test/e2e/framework/scaletest/get-publish-metrics.go

```diff
@@ -0,0 +1,212 @@
+package scaletest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/microsoft/retina/pkg/telemetry"
+	"github.com/microsoft/retina/test/e2e/common"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	metrics "k8s.io/metrics/pkg/client/clientset/versioned"
+)
+
+type GetAndPublishMetrics struct {
+	KubeConfigFilePath          string
+	AdditionalTelemetryProperty map[string]string
+	Labels                      map[string]string
+	OutputFilePath              string
+	stop                        chan struct{}
+	wg                          sync.WaitGroup
+	telemetryClient             *telemetry.TelemetryClient
+	appInsightsKey              string
+}
+
+func (g *GetAndPublishMetrics) Run() error {
+	if g.appInsightsKey != "" {
+		telemetry.InitAppInsights(g.appInsightsKey, g.AdditionalTelemetryProperty["retinaVersion"])
+
+		telemetryClient, err := telemetry.NewAppInsightsTelemetryClient("retina-scale-test", g.AdditionalTelemetryProperty)
+		if err != nil {
+			return errors.Wrap(err, "error creating telemetry client")
+		}
+
+		g.telemetryClient = telemetryClient
+	}
+
+	g.stop = make(chan struct{})
+	g.wg.Add(1)
+
+	go func() {
+
+		t := time.NewTicker(5 * time.Minute)
+
+		for {
+			select {
+
+			case <-t.C:
+				err := g.getAndPublishMetrics()
+				if err != nil {
+					log.Fatalf("error getting and publishing number of restarts: %v", err)
+					return
+				}
+
+			case <-g.stop:
+				g.wg.Done()
+				return
+
+			}
+		}
+
+	}()
+
+	return nil
+}
+
+func (g *GetAndPublishMetrics) Stop() error {
+	telemetry.ShutdownAppInsights()
+	close(g.stop)
+	g.wg.Wait()
+	return nil
+}
+
+func (g *GetAndPublishMetrics) Prevalidate() error {
+	if os.Getenv("AZURE_APP_INSIGHTS_KEY") == "" {
```

Comment:
env's as strings can be hard to track down if changed, maybe put this as a common variable

Reply:
Done